packages/mcl/src/src/mcl/commands/config.d

```diff
@@ -14,112 +17,166 @@ import mcl.utils.nix : nix, queryStorePath;
+struct start_vm_args
+{
+    @(PositionalArgument(0).Optional().Placeholder("vm-name").Description("Name of the VM"))
+    string vmName = "";
+}
+
+@(Command(" ").Description(" "))
+struct unknown_command_args
+{
+}
+
+int unknown_command(unknown_command_args unused)
+{
+    errorAndExit("Unknown command. Use --help for a list of available commands.");
+    return 1;
+}
 
-    string subcommand = args.front;
-
-    switch (subcommand) {
-        case "sys":
-            sys(args.drop(1));
-            break;
-        case "home":
-            home(args.drop(1));
-            break;
-        case "start-vm":
-            startVM(args.drop(1));
-            break;
-        default:
-            errorAndExit("Unknown config subcommand " ~ subcommand ~ ". Supported subcommands: sys, home, start-vm");
-            break;
+export int config(config_args args)
+{
+    if (!checkRepo())
+    {
+        errorAndExit(
+            "This command must be run from a repository containing a NixOS machine configuration");
     }
+
+    return args.cmd.match!(
+        (sys_args a) => sys(a),
+        (home_args a) => home(a),
+        (start_vm_args a) => startVM(a.vmName),
+        (unknown_command_args a) => unknown_command(a)
+    );
 }
 
 bool checkRepo()
 {
     const string[] validRepos = ["nixos-machine-config", "infra-lido"];
-    string remoteOriginUrl = execute(["git", "config", "--get", "remote.origin.url"], false);
+    string remoteOriginUrl = execute([
+        "git", "config", "--get", "remote.origin.url"
+    ], false);
 
-    foreach (string repo; validRepos) {
-        if (remoteOriginUrl.indexOf(repo) != -1) {
+    foreach (string repo; validRepos)
+    {
+        if (remoteOriginUrl.indexOf(repo) != -1)
+        {
             return true;
         }
     }
     return false;
 }
 
-void executeCommand(string command) {
+int executeCommand(string command)
+{
     auto exec = execute!ProcessPipes(command, true, false, Redirect.stderrToStdout);
-    wait(exec.pid);
+    return wait(exec.pid);
 }
 
-void edit(string type, string path) {
+int edit(string type, string path)
+{
     string editor = environment.get("EDITOR", "vim");
     string user = environment.get("USER", "root");
     writeln("Editing " ~ path ~ " configuration from: ", path);
-    final switch (type) {
-        case "system":
-            executeCommand(editor ~ " machines/*/" ~ path ~ "/*.nix");
-            break;
-        case "user":
-            executeCommand(editor~ " users/" ~ user ~ "/gitconfig " ~ "users/" ~ user ~ "/*.nix " ~ "users/" ~ user ~ "/home-"~path~"/*.nix");
-            break;
+    final switch (type)
+    {
+    case "system":
+        return executeCommand(editor ~ " machines/*/" ~ path ~ "/*.nix");
+    case "user":
+        return executeCommand(editor ~ " users/" ~ user ~ "/gitconfig " ~ "users/" ~ user ~ "/*.nix " ~ "users/" ~ user ~ "/home-" ~ path ~ "/*.nix");
     }
 }
 
-void sys(string[] args)
+int apply(string type, string value)
```

Comment:
I suggest having two separate functions:

```d
int switchSystem(string machineName);
int switchHome(HomeConfigType type, string username);
```

The comment about `switchHome` is dependent on this change in the Justfile:

<img width="438" height="97" alt="Image" src="https://github.com/user-attachments/assets/997de006-d069-4452-8d20-c9b523fa5228" />